### ISSUE TITLE
Adding Prometheus metric type comments

### ISSUE THREAD

**Issue by eMCeee89** [IntersectMBO/cardano-node#5140]
**Internal/External**
*External*

**Area**
*Other*

**Summary**
There are metric types missing in Prometheus metrics endpoint. It would be great to have them added for all the metrics. There is a problem if we want to scrape them using AWS CloudWatch agent because metrics without type are simply dropped and [a feature request for adding these types explicitly as a workaround](https://github.com/aws/amazon-cloudwatch-agent/issues/187) has never been implemented. It would be great to have this standard included, e.g. as a node_exporter has it. I am attaching a snippet of node_exporter metrics output, it is really about an extra line for every metric stating its type (help is optional but would be nice too):
```
# curl 127.0.0.1:9100/metrics
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 1.922e-05
go_gc_duration_seconds{quantile="0.25"} 2.694e-05
go_gc_duration_seconds{quantile="0.5"} 3.0461e-05
go_gc_duration_seconds{quantile="0.75"} 7.9441e-05
go_gc_duration_seconds{quantile="1"} 7.9441e-05
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 8
...
```

**Steps to reproduce**
1. Enable prometheus metrics endpoint
2. Go to prometheus metrics endpoint and observe missing TYPE comment lines for all the metrics

**Expected behavior**
I expect to have metric types defined when querying Prometheus metrics endpoint so these can be correctly scraped by AWS CloudWatch agent.

**System info (please complete the following information):**
- OS Name: Amazon Linux 2
- OS Version: amzn2-ami-ecs-kernel-5.10-hvm
- Node version: 1.35.7
- CLI version : 1.35.7 - linux-x86_64 - ghc-8.10, git rev 0000000000000000000000000000000000000000

**Comment by jutaro** [comment]
What do you mean with **Prometheus metrics endpoint**? Cardano node emits metrics in a format that can be used by Prometheus as defined here [](https://prometheus.io/docs/concepts/data_model/).  This is an untyped format. It is described here [](https://developers.cardano.org/docs/stake-pool-course/handbook/apply-logging-prometheus/) how to setup Prometheus to process the metrics. On this page the use of  `node_exporter` is recommended. However this is a tool of the Prometheus open source infrastructure, and it is not developed by us.

**Comment by eMCeee89** [comment]
Hello, @jutaro - I meant Cardano node emitted metrics in a format that can be used by Prometheus would be more portable if they contain metric metadata. The most important one is TYPE. Such metadata is being exposed by many exporters (including the mentioned node_exporter) because many client libraries are making use of it. Have a look at e.g. Grafana article from 2020 [here](https://grafana.com/blog/2020/06/15/how-we-made-working-with-prometheus-easier-with-metric-metadata-in-grafanas-explore-view/).

So the suggested change would consist of adding TYPE metadata comment line into metrics emitted by Cardano node. This would allow scraping Cardano node metrics by AWS CloudWatch agent but perhaps more implementations miss such metadata too.

**Comment by eMCeee89** [comment]
Metric types are documented as such here: https://prometheus.io/docs/concepts/metric_types/
I found no official documentation about those metadata definitions in Prometheus docu and even in the documentation mentioned in this message, it says:
```
The Prometheus client libraries offer four core metric types. These are currently only differentiated in the client libraries (to enable APIs tailored to the usage of the specific types) and in the wire protocol. The Prometheus server does not yet make use of the type information and flattens all data into untyped time series. This may change in the future.
```

But as I wrote, it would be nice to have it so a wider range of clients can scrape Cardano node metrics. You can check the implementation of node_exporter, e.g. here is the final format: https://github.com/prometheus/node_exporter/blob/b87c6a8826d41a242182f798e3e5688c870a9b12/collector/fixtures/ip_vs_result.txt

**Comment by eMCeee89** [comment]
Hmm, once for a session? I do not know what you mean by that. Every scraping HTTP request expects metric names with their values and metadata. In the example above, there is the same metric (the same metric name) with a multiple labels but the metric type is still the same hence only one metadata of type TYPE is enough. It makes sense to add such metadata right above the first occurrence of metric but as far as I know, order is not important and you can expose those metrics including metadata in a random order.

**Comment by eMCeee89** [comment]
Ping

**Comment by mgmeier** [comment]
Implemented by #5997 for the new, `cardano-tracer` based tracing system. Will not be backported to the current (soon to be legacy) tracing system.
